src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3225,19 +3237,45 @@ QString ViewProviderSketch::getPresentationString(const Constraint *constraint)
             {
                 // Example code from: Mod/TechDraw/App/DrawViewDimension.cpp:372
                 QRegExp rxUnits(QString::fromUtf8(" \\D*$"));  //space + any non digits at end of string
-                userStr.remove(rxUnits);              //getUserString(defaultDecimals) without units
+                valueStr.remove(rxUnits);                      //getUserString(defaultDecimals) without units
             }
         }
     }
 
     if (constraint->Type == Sketcher::Diameter){
-        userStr.insert(0, QChar(8960)); // Diameter sign
+        valueStr.insert(0, QChar(8960)); // Diameter sign
     }
     else if (constraint->Type == Sketcher::Radius){
-        userStr.insert(0, QChar(82)); // Capital letter R
+        valueStr.insert(0, QChar(82)); // Capital letter R
+    }
+
+    /**
+    Create the representation string from the user defined format string
+    Format options are:
+    %N - the constraint name parameter
+    %V - the value of the dimensional constraint, including any unit characters
+    */
+    if (iShowDimName && !nameStr.isEmpty())
+    {
+        if (formatStr.contains(QLatin1String("%V")) || formatStr.contains(QLatin1String("%N")))

```

Comment:
Just wonder if QLatin1String is included if PreCompiled.h isn't there. I don't know how to test this but maybe it's better to include it in the file header.

Reply:
PreCompiled.h already is included in the file. And it should work without problem as `QString::fromLatin1()` is already used in the function on earlier commits -> [switch for unitSys](https://github.com/FreeCAD/FreeCAD/blob/b8f4a152d3b8761a13a830cf4f266e9fc42d61fc/src/Mod/Sketcher/Gui/ViewProviderSketch.cpp#L3197-L3219), which is essentially the same as `QLatin1String`.

Reply:
Precompiled is always included but sometimes not available on the system, which may lead to compilation errors that you not see on system where it is present. 
QLatin1String is really a different class from QString. It is just a wrapper around std strings.

Reply:
Agreed. 
But I would follow the advice of the [QT doc](https://doc.qt.io/qt-5/qlatin1string.html#details) and the [linked article](https://woboq.com/blog/qstringliteral.html) (which is from the guys who actually authored the `QstringLiteral` macro implementation), to use `QLatin1String` instead of `QStringLiteral` if the function has an overload for it. `.contains()`, `.replace()` and `operator+` actually do.

Reply:
It has nothing to do with the usage. I agree with that. Just I wonder if we have to add `#include <QLatin1String>` at the top of the file so it compiles correctly when Precompiled is absent. ;) 
@chennes do you have clue on this?

Reply:
The above mentioned [code section for the unitsSys switch](https://github.com/FreeCAD/FreeCAD/blob/b8f4a152d3b8761a13a830cf4f266e9fc42d61fc/src/Mod/Sketcher/Gui/ViewProviderSketch.cpp#L3197-L3219) could actually use `QStringLiteral`, as here only a `QString` is constructed, but not directly used in a function call.

Reply:
Are you both compiling on Windows? If so I'll spool up a Linux compile to see if it complains about the absence of the include file.

Reply:
I'm on macOS Big Sur 11.5.2. Compiles fine on my side.

Reply:
OK, I just tested on Linux, too, it compiles fine as-is. Adding the header does not seem to be necessary, it's probably already being included by whatever is bringing in the `QString::fromLatin1()` function (`QString.h`, presumably).